igor-web/src/main/groovy/com/netflix/spinnaker/igor/scm/stash/client/StashClient.groovy

```diff
@@ -31,5 +34,23 @@ interface StashClient {
         @Path('projectKey') String projectKey,
         @Path('repositorySlug') String repositorySlug,
         @QueryMap Map queryMap)
+
+  // TODO: pagination support
+  @GET('/rest/api/1.0/projects/{projectKey}/repos/{repositorySlug}/browse/{dirPath}')
+  DirectoryListingResponse listDirectory(
+    @Path('projectKey') String projectKey,
+    @Path('repositorySlug') String repositorySlug,
+    @Path(value = 'dirPath', encode = false) String dirPath,
+    @Query('at') String at
+  )
+
+  // TODO: pagination support
```

Comment:
This might be needed if the manifests start to grow very large, especially with delivery configs, but I don't plan to add it as part of this PR.